x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java

```diff
@@ -62,11 +65,19 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
                 restRequest.paramAsStringArray(
                     GetTrainedModelsAction.Request.INCLUDE.getPreferredName(),
                     Strings.EMPTY_ARRAY)));
-        final GetTrainedModelsAction.Request request = restRequest.hasParam(GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION) ?
-            new GetTrainedModelsAction.Request(modelId,
+        final GetTrainedModelsAction.Request request;
+        if (restRequest.hasParam(GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION)) {
+            deprecationLogger.deprecate(
+                GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION,
+                "[{}] parameter is deprecated! Use parameter [{}] and supply the flag 'definition' instead.",
+                GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION,
+                GetTrainedModelsAction.Request.INCLUDE.getPreferredName());
+            request = new GetTrainedModelsAction.Request(modelId,
```

Comment:
Should this constructor now go away?
```
        public Request(String id, boolean includeModelDefinition, List<String> tags) {
```

Reply:
No, it's a convenience method for when the user calls with a deprecated flag. Consequently, itself is also deprecated.